ncrypt/crypt.c

```diff
@@ -195,6 +195,9 @@ int mutt_protect(struct Header *msg, char *keylist)
       crypt_smime_set_sender(mailbox);
     else if ((WithCrypto & APPLICATION_PGP) && (msg->security & APPLICATION_PGP))
       crypt_pgp_set_sender(mailbox);
+
+    if (!msg->env->from)
+      FREE(&from);
   }
```

Comment:
I think this needs to `FREE(&from->personal)` too.
`mutt_default_from()` either generates a `rfc822_new_address()` or calls `rfc822_cpy_adr_real(From)`
`rfc822_cpy_adr_real()` generates a `rfc822_new_address()` and sets `->personal`